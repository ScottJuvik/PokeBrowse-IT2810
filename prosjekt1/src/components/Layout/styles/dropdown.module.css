.navbar {
    position: relative;
    width: 100%;
}

.icon {
    position: absolute;
    top: 1rem;
    right: 1rem;
    z-index: 2;
}

/* Dropdown menu styling */
.dropdown {
    position: absolute;
    background-color: rgba(0, 0, 0, 0.7);
    width: 100%;
    left: 0;
    top: -200px;
    padding: 10;
    margin: 10;
    transition: top 0.3s ease-in-out;
    z-index: 1;
}

.open {
    top: 0;
    display: flex;
    flex-direction: column;
    display: block;
}

.dropdown li {
    color: var(--text-white);
    font-size: var(--font-size-xl);
    font-weight: var(--font-weight-semibold);
    padding: 10px;
}
