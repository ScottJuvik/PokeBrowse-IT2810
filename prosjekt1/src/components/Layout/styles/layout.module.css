.layout {
    display: grid;
    grid-template-rows: auto 1fr;
    min-height: 100vh;
    width: 100%;

    nav {
        grid-row: 1;
    }

    .content {
        grid-row: 2;
        overflow-y: auto;
    }
}

.layoutMobile {
    width: 100%;
    height: 100vh;
    overflow-x: hidden;
    position: relative;

    .contentMobile {
        flex-grow: 1;
        overflow-y: auto;
        height: 100%;
        width: 100%;
    }
}
