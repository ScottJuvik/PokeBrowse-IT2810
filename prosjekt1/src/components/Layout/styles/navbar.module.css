@import '@styles/variables.css';

.navbar {
    background-color: var(--bg-primary);
    padding: 1.5rem;
    color: black;
    box-shadow:
        0 10px 15px rgba(0, 0, 0, 0.1),
        0 4px 6px rgba(0, 0, 0, 0.1);

    ul {
        display: flex;
        justify-content: center;
        gap: 2.5rem;
    }

    a {
        font-size: var(--font-size-xl);
        font-weight: var(--font-weight-semibold);
        line-height: var(--line-height-xl);

        &:hover {
            color: var(--text-highlight-hover);
        }
    }

    .active {
        color: var(--text-highlight-hover);
    }
}
