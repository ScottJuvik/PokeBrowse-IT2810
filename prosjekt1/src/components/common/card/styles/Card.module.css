@import '@styles/variables.css';

.pokemonCard {
    width: 250px;
    max-height: 300px;
    border: 2px solid #d1d5db;
    border-radius: 8px;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 0.5rem;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    cursor: pointer;
    position: relative;
    overflow: hidden;
    padding: 0 1rem;
    padding-top: 2rem;
    background-color: var(--bg-primary);

    &:hover {
        img {
            transform: scale(1.2);
        }
    }
}

.pokemonImage {
    width: 75%;
    height: auto;
    min-height: 160px;
    transition: transform 0.3s ease;
    user-select: none;
}

.pokemonDetails {
    width: 100%;
    padding: 0.5rem 0;
    border-bottom-left-radius: 0.5rem;
    border-bottom-right-radius: 0.5rem;
    text-align: left;
}

.pokemonIndex {
    font-size: var(--font-size-base);
    line-height: var(--line-height-xs);
    color: #9e9e9e;
}

.pokemonName {
    font-size: var(--font-size-lg);
    line-height: var(--line-height-base);
    font-weight: var(--font-weight-semibold);
    color: var(--text-primary);
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.badgeWrapper {
    display: flex;
    justify-content: start;
    align-items: center;
    gap: 0.25rem;
    margin-top: 5px;
    margin-bottom: 5px;
    width: 100%;
}

@media screen and (min-width: 768px) {
    .pokemonCard {
        width: 200px;
        height: 260px;

        .pokemonImage {
            min-height: 123px;
        }
    }
}
