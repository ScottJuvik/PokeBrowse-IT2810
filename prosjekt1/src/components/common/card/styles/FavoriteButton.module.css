.pokemonHeart {
    position: absolute;
    top: 0.5rem;
    right: 0.5rem;
    z-index: 1;
}

.isFavorite {
    color: red;
}

button {
    cursor: pointer;
    background-color: transparent;
    outline: none;
    border: none;
}
