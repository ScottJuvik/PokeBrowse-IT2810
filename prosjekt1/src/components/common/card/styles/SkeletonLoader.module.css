.skeletonCard {
    width: 250px;
    max-height: 290px;
    border: 2px solid #d1d5db;
    border-radius: 8px;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0 1rem;
    padding-top: 2rem;
    background-color: var(--bg-primary);
    gap: 4rem;
}

.skeletonImage {
    width: 75%;
    height: 150px;
    min-height: 160px;
    background-color: #e0e0e0;
    border-radius: 8px;
}

.skeletonDetails {
    width: 100%;
    padding: 0.5rem 0;
}

.skeletonIndex,
.skeletonName {
    height: 16px;
    background-color: #e0e0e0;
    margin-bottom: 10px;
    border-radius: 4px;
}

.skeletonIndex {
    width: 40px;
}

.skeletonName {
    width: 80%;
}

.skeletonBadges {
    height: 30px;
    width: 100%;
    background-color: #e0e0e0;
    border-radius: 4px;
}

/* Animation */
@keyframes shimmer {
    0% {
        background-position: -200px 0;
    }
    100% {
        background-position: 200px 0;
    }
}

.skeletonCard div {
    background: linear-gradient(90deg, #f0f0f0 25%, #e0e0e0 50%, #f0f0f0 75%);
    background-size: 200px 100%;
    animation: shimmer 1.2s infinite;
}

@media screen and (min-width: 768px) {
    .skeletonCard {
        width: 200px;
        height: 250px;

        .skeletonImage {
            min-height: 123px;
        }
    }
}
