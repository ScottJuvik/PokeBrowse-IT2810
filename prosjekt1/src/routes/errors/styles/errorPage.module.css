.errorContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    background-color: var(--bg-not-found);
    padding: 1.5rem;
    color: var(--text-white);

    background-color: var(--bg-error-page);

    img {
        width: 100%;
        max-width: 10rem;
    }

    h1 {
        margin-top: 1rem;
        font-weight: var(--font-weight-bold);
        font-size: var(--font-size-4xl);
        line-height: var(--line-height-4xl);
    }

    .sorryText {
        margin-top: 1rem;
        font-size: var(--font-size-4xl);
        line-height: var(--line-height-4xl);
    }

    .prankText {
        font-style: italic;
        margin-top: 0.5rem;
        font-size: var(--font-size-lg);
        line-height: var(--line-height-lg);
    }

    .errorText {
        margin-top: 0.25rem;
        font-size: var(--font-size-sm);
        line-height: var(--line-height-sm);
    }

    .statusText {
        margin-top: 1rem;
    }
}
