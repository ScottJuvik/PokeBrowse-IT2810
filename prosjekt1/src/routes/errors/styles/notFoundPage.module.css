@import '@styles/variables.css';

.errorContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    background-color: var(--bg-not-found);
    padding: 1.5rem;
    color: var(--text-white);
    text-align: center;

    img {
        width: 100%;
        max-width: 10rem;
    }

    h1,
    .subtitle {
        margin-top: 1rem;
    }

    .text {
        margin-top: 0.5rem;
    }

    button {
        border-radius: 0.375rem;
        background-color: var(--btn-not-found);
        font-weight: 600;
        border: none;
        padding: 0.5rem 1rem;
        margin-top: 1.5rem;
        transition-duration: 150ms;
        transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
        color: var(--text-primary);

        &:hover {
            background-color: var(--btn-not-found-hover);
            cursor: pointer;
        }

        &:focus {
            outline: none;
        }
    }
}
