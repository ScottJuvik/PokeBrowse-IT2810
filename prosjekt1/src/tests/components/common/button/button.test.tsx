import { render, screen, fireEvent } from '@testing-library/react';
import Button from '@components/common/button/button.tsx';
import styles from '@components/common/button/button.module.css';
import { vi } from 'vitest';

describe('Button Component', () => {
    test('renders the button with correct text', () => {
        render(<Button text="Catch 'Em All!" onClick={() => {}} />);
        const buttonElement = screen.getByText(/Catch 'Em All!/i);
        expect(buttonElement).toBeInTheDocument();
    });

    test('fires onClick event when button is clicked', () => {
        const handleClick = vi.fn();
        render(<Button text="Click Me" onClick={handleClick} />);
        const buttonElement = screen.getByText(/Click Me/i);
        fireEvent.click(buttonElement);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    test('applies primary variant class', () => {
        render(<Button text="Primary Button" onClick={() => {}} variant="primary" />);
        const buttonElement = screen.getByText(/Primary Button/i);
        expect(buttonElement).toHaveClass(styles.btn);
        expect(buttonElement).toHaveClass(styles.primary);
    });

    test('is disabled when the disabled prop is passed', () => {
        render(<Button text="Disabled Button" onClick={() => {}} disabled={true} />);
        const buttonElement = screen.getByText(/Disabled Button/i);
        expect(buttonElement).toBeDisabled();
    });

    test('applies the correct size class', () => {
        render(<Button text="Large Button" onClick={() => {}} size="large" />);
        const buttonElement = screen.getByText(/Large Button/i);
        expect(buttonElement).toHaveClass(styles.large);
    });
});
