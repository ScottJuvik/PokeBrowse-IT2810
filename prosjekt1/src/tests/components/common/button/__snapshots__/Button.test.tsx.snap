// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches the snapshot 1`] = `
<div>
  <button
    class="_btn_07d6f9 _outline_07d6f9 _medium_07d6f9"
  >
    Catch 'Em All!
  </button>
</div>
`;
